server/courselist.rb

```diff
@@ -379,19 +438,16 @@ def self.vetoRequest(user, request_url)
   ## only applies to requests for the top level Dashboard page.  This processing
   ## is off by default.
   before '/' do
-
     ## Check and possibly override the user id
-    logger.debug "#{__LINE__}:authn_uniqname_override: "+@@authn_uniqname_override.to_s
-    uniqnameOverride if @@authn_uniqname_override == true
-
+    uniqnameOverride if allow_uniqname_override request.env['REMOTE_USER']
   end
 
-  ## Check that any request for userid data is either for data for the authenticated userid
-  ## or the authenticated userid is listed as a special admin userid.
+  ## Check that any request for userid data is allowed.
   before "*" do
-    ## Make sure people only ask about themselves (or are privileged)
-    vetoResult = CourseList.vetoRequest request.env['REMOTE_USER'], request.env['REQUEST_URI']
-    logger.debug "vetoResult: "+vetoResult.to_s
+
+    # NOTE: the {} block on the end is passed in and used to see if this is an admin user.
+    vetoResult = CourseList.vetoRequest(request.env['REMOTE_USER'], request.env['REQUEST_URI']) { admin_user request.env['REMOTE_USER'] }
+    logger.debug "REQUEST: * end veto check: [#{vetoResult}]"
     halt 401 if vetoResult == true
```

Comment:
Based on the link Chris found, I think this should be 403
http://stackoverflow.com/questions/3297048/403-forbidden-vs-401-unauthorized-http-responses